### PR TITLE
Expose ExportedProgram and related classes

### PR DESCRIPTION
**Pull request by gmagogsfm**

### CODE REVIEW

**Review by zhxchen17 — APPROVED** [review]

**Review by angelayi — APPROVED** [review]

File: torch/_export/__init__.py [hunk 1]
```diff
@@ -338,14 +343,20 @@ def export(
...
+
     export_graph_signature = ExportGraphSignature(
-        parameters=graph_signature.parameters,
-        buffers=graph_signature.buffers,
-        user_inputs=graph_signature.user_inputs,
-        user_outputs=graph_signature.user_outputs,
-        inputs_to_parameters=graph_signature.inputs_to_parameters,
-        inputs_to_buffers=graph_signature.inputs_to_buffers,
-        buffers_to_mutate=graph_signature.buffers_to_mutate,
+        parameters=to_str_list(graph_signature.parameters),
```
Comment (avikchaudhuri) [inline torch/_export/__init__.py:353]: What's the `to_str_list` / `to_str_dict` doing? Why is it necessary? Does it need to be parsed back to non-str objects?

Comment (gmagogsfm) [reply torch/_export/__init__.py:353]: The original types are aot autograd internal types that just alias str. I am not comfortable exposing internal types that can change on our public interface. 

They don't need to be converted back I think

**Comment by qihqi** [comment]
QQ: why not leave the exported_program.py file along and just add an alias in `torch/export/__init__.py`

**Pull request closed without merging**